Server/controllers/car.js

```diff
@@ -39,18 +39,26 @@ class carController {
         try {
             const { car_id, amount, status } = req.body;
             const buyer = jwt.decode(req.body.token).email;
-            const singleCar = await runQuery(car.getCar, [car_id]);
-            if (!singleCar[0] || singleCar[0].owner === buyer) throw new Error('Owner can not make a purchase order');
+            const isOrderOwner = await runQuery(car.carOwner, [car_id, buyer]);
             const isOrderExist = await runQuery(order.isOrderExist, [car_id, buyer]);
-            if (isOrderExist[0]) throw new Error("order is already exist");
-            const newOrder = [buyer, car_id, amount];
-            const result = await runQuery(order.createOrder, newOrder);
-            res.status(201).json({
-                status: 201,
-                data: result[0]
-            });
+
```

Comment:
You can refactor this to use less nested if statements which can make your code hard to read. I recommend using a fail-fast approach, first check the errors and return.

e.g.
```
if (isOrderOwner[0]) throw new Error('Owner can not make a purchase order');
if (isOrderExist[0]) throw new Error('order is already exist');
const newOrder = [buyer, car_id, amount];
const result = await runQuery(order.createOrder, newOrder);
res.status(201).json({
    status: 201,
    data: result
});
```